app/channel_test.go

```diff
@@ -1657,6 +1658,67 @@ func TestPatchChannelModerationsForChannel(t *testing.T) {
 			}
 		})
 	}
+
+	t.Run("Handles concurrent patch requests gracefully", func(t *testing.T) {
+		wg := sync.WaitGroup{}
+		wg.Add(20)
+		for i := 0; i < 10; i++ {
+			go func() {
+				th.App.PatchChannelModerationsForChannel(channel, []*model.ChannelModerationPatch{
+					{
+						Name: &createPosts,
+						Roles: &model.ChannelModeratedRolesPatch{
+							Members: model.NewBool(true),
+							Guests:  model.NewBool(true),
+						},
+					},
+				})
+				th.App.PatchChannelModerationsForChannel(channel, []*model.ChannelModerationPatch{
+					{
+						Name: &createPosts,
+						Roles: &model.ChannelModeratedRolesPatch{
+							Members: model.NewBool(false),
+							Guests:  model.NewBool(false),
+						},
+					},
+				})
+				wg.Done()
+			}()
+		}
+
+		for i := 0; i < 10; i++ {
+			go func() {
+				th.App.PatchChannelModerationsForChannel(channel, []*model.ChannelModerationPatch{
+					{
+						Name: &createPosts,
+						Roles: &model.ChannelModeratedRolesPatch{
+							Members: model.NewBool(true),
+							Guests:  model.NewBool(true),
+						},
+					},
+				})
+				th.App.PatchChannelModerationsForChannel(channel, []*model.ChannelModerationPatch{
+					{
+						Name: &createPosts,
+						Roles: &model.ChannelModeratedRolesPatch{
+							Members: model.NewBool(false),
+							Guests:  model.NewBool(false),
+						},
+					},
+				})
+				wg.Done()
+			}()
+		}
+
+		wg.Wait()
+
+		higherScopedGuestRoleName, higherScopedMemberRoleName, _, _ := th.App.GetTeamSchemeChannelRoles(channel.TeamId)
+		higherScopedMemberRole, _ := th.App.GetRoleByName(higherScopedMemberRoleName)
+		higherScopedGuestRole, _ := th.App.GetRoleByName(higherScopedGuestRoleName)
+		assert.Contains(t, higherScopedMemberRole.Permissions, createPosts)
+		assert.Contains(t, higherScopedGuestRole.Permissions, createPosts)
```

Comment:
@agnivade I added this test and confirmed that it fails consistently on master (without the fix in this PR) and that once I switch to this PR it was passing consistently